app/assets/javascripts/school_administrator_dashboard/page_container.jsx

```diff
@@ -0,0 +1,122 @@
+    attendanceData: React.PropTypes.object.isRequired  //should rename more generically
+  },
+
+  //necessary?
+  // studentAbsences: function () {
+  //   const students = this.props.attendanceData;
+
+  // },
+
+  //returns map of homerooms to absence events
+  absencesByHomeroom: function() {
+    //Group data by homeroom
+    const homeRoomGroups = _.groupBy(this.props.attendanceData.absences, 'homeroom');
+    let homeroomAbsences = {};
+    //Simplify homeroom data into map of homerooms to total homeroom absences along with homeroom size
+    Object.keys(homeRoomGroups).forEach( (key) => {
+      const size = homeRoomGroups[key].length;
+      let key2 = key;
+      if (key == "null") key2 = "No Homeroom";
+      homeroomAbsences[key2] = {
+        //Group events into day buckets
+        absences: this.eventsGroupedByDay(_.flattenDeep(_.map(homeRoomGroups[key], function(student) {
+          return student.absences;
+        }))),
+        size: size
+      };
+    });
+    return homeroomAbsences;
+  },
+
+  //Returns a list of homerooms along with dates and attendance percentage for each
+  attendanceByHomeroom: function () {
+    let homeroomDays = this.absencesByHomeroom();
+    Object.keys(homeroomDays).forEach((homeroom) => {
+      homeroomDays[homeroom].absences = this.averageDailyAttendance(homeroomDays[homeroom].absences, homeroomDays[homeroom].size);
+    });
+    return homeroomDays;
+  },
+
+  //Returns list of dates along with attendance percentage for all active students in school
+  attendanceBySchool: function() {
+    let schoolAbsences = {};
+    const homerooms = this.absencesByHomeroom();
+    Object.keys(homerooms).forEach((key) => {
+      this.mergeDateLists(schoolAbsences, homerooms[key].absences);
+    });
+    const schoolAttendance = this.averageDailyAttendance(schoolAbsences, this.totalStudents());
+    return schoolAttendance;
+  },
+
+  //Returns average monthly attendance givenn set of daily averages. To be used after applying date filters.
+  monthlyAttendanceBySchool: function () {
+    const schoolAttendance = this.attendanceBySchool();
+    let monthlyAttendance = {};
+    Object.keys(schoolAttendance).forEach((day) => {
+      if (monthlyAttendance[moment(day).date(1).format("YYYY-MM-DD")] === undefined) {
+        monthlyAttendance[moment(day).date(1).format("YYYY-MM-DD")] = [schoolAttendance[day]];
+      } else {
+        monthlyAttendance[moment(day).date(1).format("YYYY-MM-DD")].push(...schoolAttendance[day]);
+      }
+    });
+    return monthlyAttendance;
+  },
+
+  averageDailyAttendance: function(dailyAbsences, size) {
+    let averageDailyAttendance = {};
+    //mapping each homeroom to an array of day buckets containing all absences for each day w/in the homeroom
+    Object.keys(dailyAbsences).forEach((day) => {
+      averageDailyAttendance[day] = (size - dailyAbsences[day].length)/size*100;
+    });
+    return averageDailyAttendance;
+  },
+
+  mergeDateLists: function(list1, list2) {
```

Comment:
Could we use lodash `_.merge` instead of writing our own merge? @edavidsonsawyer

Reply:
For some reason that didn't work the way I expected it to. I'll look at it again and see about simplifying this.